ambari-server/src/main/java/org/apache/ambari/server/state/DependencyInfo.java

```diff
@@ -39,6 +40,7 @@
    * The name of the component which is the dependency.
    * Specified in the form serviceName/componentName.
    */
+  @XmlTransient // bound via setter, to also set serviceName and componentName
   private String name;
```

Comment:
It seem to be inconsistent to declare @XmlAccessorType(XmlAccessType.FIELD) on the class level meaning properties should be set on the field level, then making the fields @XmlTransient. I would suggest removing the class level @XmlAccessorType annotation instead.

Reply:
Not all fields are `@XmlTransient`, only the exceptions.  [Javadoc for `XmlAccessType`](https://docs.oracle.com/javaee/6/api/javax/xml/bind/annotation/XmlAccessType.html) implies this is valid.